src/mds/MDSAuthCaps.h

```diff
@@ -35,23 +33,8 @@ enum {
   MAY_SNAPSHOT	= (1 << 7),
 };
 
-class CephContext;
-
 // what we can do
 struct MDSCapSpec {
-  static const unsigned ALL		= (1 << 0);
-  static const unsigned READ		= (1 << 1);
-  static const unsigned WRITE		= (1 << 2);
-  // if the capability permits setting vxattrs (layout, quota, etc)
-  static const unsigned SET_VXATTR	= (1 << 3);
-  // if the capability permits mksnap/rmsnap
-  static const unsigned SNAPSHOT	= (1 << 4);
-
-  static const unsigned RW		= (READ|WRITE);
-  static const unsigned RWP		= (READ|WRITE|SET_VXATTR);
-  static const unsigned RWS		= (READ|WRITE|SNAPSHOT);
-  static const unsigned RWPS		= (READ|WRITE|SET_VXATTR|SNAPSHOT);
```

Comment:
IIRC, static members should come before constructors.

Reply:
Yes. I have updated the patch.